sbi/neural_nets/density_estimators/categorical_net.py

```diff
@@ -91,7 +93,9 @@ def log_prob(self, input: Tensor, context: Tensor) -> Tensor:
         """
         # Predict categorical ps and evaluate.
         ps = self.forward(context)
-        return Categorical(probs=ps).log_prob(input.squeeze())
+        # Squeeze dim=1 because `Categorical` has `event_shape=()` but our data usually
+        # has an event_shape of `(1,)`.
+        return Categorical(probs=ps).log_prob(input.squeeze(dim=1))
```

Comment:
This will break if the input is of shape (1,1), which would work using the previous code.

Reply:
The `CategoricalMassEstimator` is a `DensityEstimator` with `event_shape=()` and it passes all tests. Not sure if we need to support `(1,1)` shape.